src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -306,15 +290,44 @@ static bool IsPropertyWithInaccessibleImplementableAccessor(ISymbol member, ISym
 
                 var property = (IPropertySymbol)member;
 
-                return IsInaccessibleImplementableAccessor(property.GetMethod, within) || IsInaccessibleImplementableAccessor(property.SetMethod, within);
+                return IsNonPublicImplementableAccessor(property.GetMethod) || IsNonPublicImplementableAccessor(property.SetMethod);
             }
 
-            static bool IsInaccessibleImplementableAccessor(IMethodSymbol? accessor, ISymbol within)
+            static bool IsNonPublicImplementableAccessor(IMethodSymbol? accessor)
             {
-                return accessor != null && IsImplementable(accessor) && !accessor.IsAccessibleWithin(within);
+                return accessor != null && IsImplementable(accessor) && accessor.DeclaredAccessibility != Accessibility.Public;
             }
         }
 
+        private static ImmutableArray<ISymbol> GetExplicitlyImplementableMembers(INamedTypeSymbol type, ISymbol within)
+        {
+            if (type.TypeKind == TypeKind.Interface)
+            {
+                return type.GetMembers().WhereAsArray(m => m.Kind != SymbolKind.NamedType &&
+                                                           IsImplementable(m) && m.IsAccessibleWithin(within) &&
+                                                           !IsPropertyWithInaccessibleImplementableAccessor(m, within));
+            }
+
+            return type.GetMembers();
+        }
+
+        private static bool IsPropertyWithInaccessibleImplementableAccessor(ISymbol member, ISymbol within)
+        {
+            if (member.Kind != SymbolKind.Property)
+            {
+                return false;
+            }
+
+            var property = (IPropertySymbol)member;
+
+            return IsInaccessibleImplementableAccessor(property.GetMethod, within) || IsInaccessibleImplementableAccessor(property.SetMethod, within);
+        }
+
+        private static bool IsInaccessibleImplementableAccessor(IMethodSymbol? accessor, ISymbol within)
+        {
+            return accessor != null && IsImplementable(accessor) && !accessor.IsAccessibleWithin(within);
+        }
```

Comment:
i haven't really reviews the changes in the file.  but i'm curious why all these changes were necessary.  what was the overall motivation for the changes here?

Reply:
These were local functions in GetAllUnimplementedExplicitMembers, but now we reuse them in GetAllUnimplementedMembers, so we had to bring them out.

Reply:
I think the order of the methods got mucked up in the process of all the refactoring and name changing. I'll try to fix it.